tests/tests_pytorch/strategies/launchers/test_subprocess_script.py

```diff
@@ -48,7 +50,7 @@ def task_fn(cfg):
 
 @RunIf(min_cuda_gpus=2, skip_windows=True, standalone=True)
 @pytest.mark.skipif(not _HYDRA_WITH_RUN_PROCESS, reason=str(_HYDRA_WITH_RUN_PROCESS))
-@pytest.mark.parametrize("subdir", [None, "dksa", ".hello"])
+@pytest.mark.parametrize("subdir", [None, "null", "dksa", ".hello"])
```

Comment:
Is "null" simulating the same we already have under the None case?

Reply:
No, these are different. Subdir being `"null"` leads to passing `hydra.output_subdir=null` to the subprocess, which leads to `output_subdir` being `None` inside the subprocess, which leads to there being no directory to store `config.yaml` etc. (Which is the NeMo use-case).

Subdir being `None`, on the other hand, leads to the check in line 65, according to which we don't pass any `hydra.output_subdir` term, which leads to `output_subdir` having its default value (`.hydra`) inside the subprocess.

Having written this out, it sounds pretty confusing. Not sure what we could change it to, though. Open to suggestions.

Reply:
Thanks for the explanation. Since the rundir="null" comes from the hydra configuration of the user, this should be fine and it is not breaking any behavior. 

Since me and others from the core team can't say much about this topic (we're not hydra users), I'm fine with this. Whatever you think is intuitive and convenient for the typical Hydra user. 

What's important for us is that we don't create undesired breaking changes for large repos like NeMo which heavily use hydra. But you already commented about this and so from my side, good to move forward with this.